### PR TITLE
[SYCL][DevOps]  Fuse OCL CPU testing into GEN9 Linux task

### PR DESCRIPTION
**Pull request by aelovikov-intel**
It would increase its duration by several minutes while eliminating 20+ minutes task completely.

### CODE REVIEW

File: devops/test_configs.json [hunk 1]
```diff
@@ -2,14 +2,24 @@
...
+      "cmake_args": ""
+    },
+    {
+      "config": "win_gen12",
+      "name": "SYCL E2E on Intel CPU/GEN12 GPU",
+      "runs-on": [
+        "Windows",
+        "gen12"
+      ],
+      "targets": "ext_oneapi_level_zero:gpu;opencl:gpu;opencl:cpu",
```
Comment (aelovikov-intel) [inline devops/test_configs.json:22]:
```suggestion
      "targets": "ext_oneapi_level_zero:gpu;opencl:gpu",
```

File: sycl/test-e2e/Config/allowlist.cpp [hunk 2]
```diff
@@ -1,14 +1,16 @@
-// REQUIRES: cpu
+// REQUIRES: opencl && cpu
```
Comment (aelovikov-intel) [inline sycl/test-e2e/Config/allowlist.cpp:1]: The change in this file seems to be effectively NFC in comparison to what we've been testing before this PR.

File: devops/test_configs.json [hunk 3]
```diff
@@ -2,14 +2,14 @@
   "lts": [
     {
       "config": "lin_gen9",
```
Comment (bader) [inline devops/test_configs.json:4]:
```suggestion
      "config": "lin_intel",
```

File: sycl/test-e2e/Config/allowlist.cpp (hunk 2 above)
Comment (bader) [reply sycl/test-e2e/Config/allowlist.cpp:1]: Why do we need this change?

File: devops/test_configs.json (hunk 3 above)
Comment (bader) [reply devops/test_configs.json:4]: or `lin_opencl`

File: sycl/test-e2e/Config/allowlist.cpp (hunk 2 above)
Comment (aelovikov-intel) [reply sycl/test-e2e/Config/allowlist.cpp:1]: The test currently fails when running on a system with devices other than opencl. 

That is possibly an issue, but even if so, it has nothing to do with this change.

Comment (bader) [reply sycl/test-e2e/Config/allowlist.cpp:1]: Please, file a bug report. AFAIK, only OpenCL backend exposes CPU devices. I expect current version to work in our CI environment.

Comment (aelovikov-intel) [reply sycl/test-e2e/Config/allowlist.cpp:1]: https://github.com/intel/llvm/issues/9551

**Comment by bader** [comment]
> It would increase its duration by several minutes while eliminating 20+ minutes task completely.

"several minutes" - could you be more specific, please?

I think it's worth noting that in your pre-commit the task took < 12 minutes (https://github.com/intel/llvm/actions/runs/5048597334/jobs/9057922308). 
Another observation: separate workflow is usually executed on runners with AMD GPU, whereas "merged workflow" can only be executed on runners with Intel GPU (which are already the main CI system bottleneck).

**Comment by aelovikov-intel** [comment]
> "several minutes" - could you be more specific, please?

33 vs 30 minutes

> I think it's worth noting that in your pre-commit the task took < 12 minutes

That was helpful. Turns out that we have `x86-cpu` property set on both AMD/Intel GEN9 runners. When dispatched onto a GEN9 runner it takes 20 minutes. And that happens when CI is already slow, and we start putting even more tasks from AMD onto GEN9 runners.

As such, it would be even more helpful than I originally thought for the times with peak loads.

> Another observation: separate workflow is usually executed on runners with AMD GPU, whereas "merged workflow" can only be executed on runners with Intel GPU (which are already the main CI system bottleneck).

I think that by decreasing the load on AMD runners we will get more heavy build jobs dispatched onto them, freeing our GEN9 runners just for this fused task.

**Review by bader — APPROVED** [review]
CI scripts changes look okay to me.

**Review by bso-intel — APPROVED** [review]

**Pull request merged**